### PR TITLE
TF-2196 Add field 'From' to composer

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
---
#2196 

## Resolved
---
- Web:

https://github.com/linagora/tmail-flutter/assets/80142234/aa6bd199-fae3-4fb0-9009-81dc4787e001

- Mobile:

https://github.com/linagora/tmail-flutter/assets/80142234/2e2b50eb-280c-4c76-b0d1-db08a02c2a28

### CODE REVIEW

**Comment by hoangdat** [comment]
- please demo case: settings identity during compose

File: lib/features/composer/presentation/composer_controller.dart [hunk 1]
```diff
@@ -552,10 +568,13 @@ class ComposerController extends BaseController {
   void _handleGetAllIdentitiesSuccess(GetAllIdentitiesSuccess success) async {
     final listIdentitiesMayDeleted = success.identities?.toListMayDeleted() ?? [];
     if (listIdentitiesMayDeleted.isNotEmpty) {
+      listFromIdentities.value = listIdentitiesMayDeleted;
       await _selectIdentity(listIdentitiesMayDeleted.first);
     }
 
-    _autoFocusFieldWhenLauncher();
+    if (fromRecipientState.value != PrefixRecipientState.enabled) {
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:575]:
```suggestion
    if (fromRecipientState.value == PrefixRecipientState.disable) {
```

File: lib/features/composer/presentation/composer_controller.dart [hunk 2]
```diff
@@ -2135,4 +2157,56 @@ class ComposerController extends BaseController {
       }
     }
   }
+
+  void onChangeIdentity (Identity? newIdentity) async {
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:2161]:
```suggestion
  void onChangeIdentity(Identity? newIdentity) async {
```

File: lib/features/composer/presentation/composer_controller.dart [hunk 3]
```diff
@@ -2135,4 +2157,56 @@ class ComposerController extends BaseController {
...
+    await _selectIdentity(newIdentity);
+  }
+
+  void onTapEditIdentity (BuildContext context) async {
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:2165]:
```suggestion
  void onTapEditIdentity(BuildContext context) async {
```

File: lib/features/composer/presentation/widgets/mobile/from_composer_mobile_widget.dart [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                                if (selectedIdentity!.name!.isNotEmpty)
+                                  TextSpan(
+                                    text: '${selectedIdentity!.name} ',
+                                    style: FromComposerMobileWidgetStyle.buttonTitleTextStyle,
+                                  ),
+                                  TextSpan(
+                                    text: '(${selectedIdentity!.email})',
+                                    style: selectedIdentity!.name!.isNotEmpty
+                                    ? FromComposerMobileWidgetStyle.buttonSubTitleTextStyle
+                                    : FromComposerMobileWidgetStyle.buttonTitleTextStyle
```
Comment (dab246) [inline lib/features/composer/presentation/widgets/mobile/from_composer_mobile_widget.dart:81]: indent

**Comment by hoangdat** [comment]
in mobile, we found a litte unnecessary space 

Screenshot 2023-11-06 at 08 54 48

**Comment by hieutbui** [comment]
> in mobile, we found a litte unnecessary space
> 
> Screenshot 2023-11-06 at 08 54 48

I changed it. Please watch the demo video above!

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Review by dab246 — APPROVED** [review]

**Pull request merged**